lang/c++/impl/ValidSchema.cc

```diff
@@ -145,49 +140,49 @@ ValidSchema::toFlatList(std::ostream &os) const
  * in UTF-8 format. Note that this method is not responsible for validating
  * the schema.
  */
-string ValidSchema::compactSchema(const string& schema) {
-    bool insideQuote = false;
-    size_t newPos = 0;
-    string data(schema.data());
-
-    for (size_t currentPos = 0; currentPos < schema.size(); currentPos++) {
-        if (!insideQuote && std::isspace(data[currentPos])) {
-            // Skip the white spaces outside quotes.
-            continue;
-        }
+    string ValidSchema::compactSchema(const string &schema) {
+        auto insideQuote = false;
+        auto newPos = 0;
+        auto data = string{schema.data()};
```

Comment:
the prior line of code actually appears more idiomatic to me?

Reply:
Yes. Changing it back.

Reply:
`string{schema.data};` seems to be what Meyer's would advocate more but still seems like extra typing to me.